File: src/clojure/earlgrey2d/audiomanager.clj
(ns earlgrey2d.audiomanager
  (:import (com.badlogic.gdx.audio Music Sound)
           (com.badlogic.gdx Gdx)))

(def cur-music (atom []))
(def cur-sound (atom []))

(defn bgm! [str]
  (if @cur-music
    (doto ^Music @cur-music
      (.stop)
      (.dispose)))
  (let [m (.newMusic Gdx/audio (.internal Gdx/files str))]
    (do
      (.play m)
      (.setLooping m true)
      (reset! cur-music m))))

(defn se! [str]
  (if @cur-sound
    (doto ^Sound @cur-sound
      (.dispose)))
  (let [s (.newSound Gdx/audio (.internal Gdx/files str))]
    (do
      (.play s)
      (reset! cur-sound s))))

File: src/java/xyz/unfound/earlgrey2d/AudioManager.java
package xyz.unfound.earlgrey2d;


import clojure.lang.Keyword;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.audio.Music;
import com.badlogic.gdx.audio.Sound;
import com.badlogic.gdx.files.FileHandle;

import java.util.HashMap;
import java.util.Stack;

public class AudioManager {
    Stack<Music> musics;
    Stack<Sound> sounds;

    public void playBgm(FileHandle fh) {
        Music music = Gdx.audio.newMusic(fh);
        musics.add(music);
        music.play();
    }

    public void playMusic(FileHandle fh) {
        Music music = Gdx.audio.newMusic(fh);
        musics.add(music);
        music.setLooping(true);
        music.play();
    }

    public void playSound(FileHandle sd) {
        
    }

    public void changeBgm(FileHandle fh) {
        Music m = musics.pop();
        m.stop();
        m.dispose();
        playBgm(fh);
    }
}